File: TemporaryMqLibrary/src/main/java/cattt.temporary.mq/base/MqBinder.java
package cattt.temporary.mq.base;

import android.os.Binder;

public class MqBinder extends Binder {
    private MqService mService;
    public MqBinder(MqService service){
        this.mService = service;
    }

    public void startConnect(){
        mService.startConnect();
    }

    public void publishMessage(String topic, String message){
        mService.publishMessage(topic, message);
    }
    public void subscribe(String[] topic){
        mService.subscribe(topic);
    }

    public boolean isConnected(){
        return mService.isConnected();
    }
}

File: CattMqKtLibs/src/main/java/catt.kt.libs.mq/core/MqBinder.kt
package catt.kt.libs.mq.core

import android.os.Binder

internal class MqBinder(private val _presenter: IConnectionPresenter) : Binder() {
    inline val isConnected: Boolean
        get() = _presenter.isConnected()

    fun connect() = _presenter.connect()

    fun publishMessage(topic: String, message: String) = _presenter.publishMessage(topic, message)

    fun subscribe(topics: Array<String>) = _presenter.subscribe(topics)
}
